Sources/TensorFlow/Core/TensorShape.swift

```diff
@@ -73,45 +73,98 @@ public struct TensorShape: ExpressibleByArrayLiteral {
     }
 }
 
-public extension TensorShape {
+extension TensorShape: Collection, MutableCollection {
+    public typealias Element = Int
+    public typealias Index = Int
+    
     /// The rank of the shape (i.e. the number of dimensions).
     @inlinable
-    var count: Int {
+    public var count: Int {
         return dimensions.count
     }
 
     @inlinable
-    var indices: Range<Int> {
+    public var indices: Indices {
         return dimensions.indices.lowerBound ..< dimensions.indices.upperBound
     }
 
     @inlinable
-    var startIndex: Int {
+    public var startIndex: Index {
         return dimensions.startIndex
     }
 
     @inlinable
-    var endIndex: Int {
+    public var endIndex: Index {
         return dimensions.endIndex
     }
-
+    
+    @inlinable
+    public func index(after i: Index) -> Index {
+        return dimensions.index(after: i)
+    }
+    
     /// Access the size of the i-th dimension.
-    /// - Parameter index: The index of a dimension.
+    /// - Parameter position: The index of a dimension.
     @inlinable
-    subscript(index: Int) -> Int {
-        _read { yield dimensions[index] }
-        _modify { yield &dimensions[index] }
+    public subscript(position: Index) -> Element {
+        _read { yield dimensions[position] }
+        _modify { yield &dimensions[position] }
     }
 
     /// Access the size of the i-th dimension.
     /// - Parameter index: The index of a dimension.
     @inlinable
-    subscript(bounds: Range<Int>) -> TensorShape {
+    public subscript(bounds: Range<Int>) -> TensorShape {
         get { return TensorShape(dimensions[bounds]) }
         set { dimensions[bounds] = ArraySlice(newValue.dimensions) }
     }
 }
 
+extension TensorShape: RandomAccessCollection {
+    public typealias Indices = Range<Int>
```

Comment:
Please move `typealias Indices = Range<Int>` to the `extension TensorShape: Collection, MutableCollection` above, since `Indices` is an associated type of `Collection`.

Reply:
done